### PR TITLE
New release process (part 5)

### PR DESCRIPTION
**Pull request by jprochazk**
### What

- Adds the `.github/actions/deploy-vercel` javascript action. It updates the target commit (`RELEASE_COMMIT`) env var, and then triggers a re-deploy of the latest production deployment.
- Updates `.github/workflows/reusable_deploy_docs` to use `deploy-vercel` if `UPDATE_LATEST` is true.
- Finalizes `release.yml` by un-disabling all the steps after `version`, together with some extra polish.
- Updates `RELEASES.md` with the new release process.

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/2992) (if applicable)

- [PR Build Summary](https://build.rerun.io/pr/2992)
- [Docs preview](https://rerun.io/preview/pr%3Ajan%2Fnew-release-process-5/docs)
- [Examples preview](https://rerun.io/preview/pr%3Ajan%2Fnew-release-process-5/examples)

### CODE REVIEW

File: .github/actions/deploy-vercel/client.mjs [hunk 1]
```diff
@@ -0,0 +1,169 @@
+// @ts-check
```
Comment (emilk) [inline .github/actions/deploy-vercel/client.mjs:1]: for this and the other `mjs` files: what do they do? Did you write them, or copy them from somewhere?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -55,21 +57,31 @@ jobs:
       - name: Update crate versions
         id: versioning
         run: |
+
+          # parse the release version from the branch name
+          #   `release-0.8.1` -> `0.8.1`
           release_version=$(echo ${{ github.ref_name }} | grep -oP '^release-\K\d+\.\d+\.\d+$')
```
Comment (emilk) [inline .github/workflows/release.yml:63]: What is the `-P` flag to grep? The grep on my mac (`grep (BSD grep, GNU compatible) 2.6.0-FreeBSD`) doesn't have it :/

File: RELEASES.md [hunk 3]
```diff
@@ -39,116 +39,40 @@ We tag all data files (`.rrd` files) and communication protocols with the rerun
...
+5. ### Run the [release workflow](https://github.com/rerun-io/rerun/actions/workflows/release.yml).
+
+   Set `Use workflow from` to the new release branch, and choose `prerelease` in the drop-down menu.
+
+   ![Image showing the Run workflow UI. It can be found at https://github.com/rerun-io/rerun/actions/workflows/release.yml](https://github.com/rerun-io/rerun/assets/1665677/6cdc8e7e-c0fc-4cf1-99cb-0749957b8328)
+
+6. ### Wait for the workflow to finish
+
+   The workflow will create a pull request for the release.
+   The PR description will contain next steps.
```
Comment (emilk) [inline RELEASES.md:78]: Very nice instructions!

**Review by emilk — APPROVED** [review]
Very nice!

The `.mjs` could do with some top-level explanations though

File: .github/actions/deploy-vercel/client.mjs (hunk 1 above)
Comment (jprochazk) [reply .github/actions/deploy-vercel/client.mjs:1]: I wrote these, it's just a bit of JS acting as a bridge between the Vercel REST API and the GitHub Actions environment. I'll add a bit of documentation, but most of the API calls are so severely underdocumented that linking to the official docs wouldn't help anyone trying to understand it 😢

File: .github/workflows/release.yml (hunk 2 above)
Comment (jprochazk) [reply .github/workflows/release.yml:63]: It makes grep use the perl-compatible regex (PCRE) engine. I don't know why the default installation of grep on mac doesn't support it. You can install the homebrew version (`brew install grep`) which will be available as `ggrep` (https://stackoverflow.com/a/22704387), and does support it

**Pull request merged**